### ISSUE TITLE
pasting lags when working on remote workspace

### ISSUE THREAD

**Issue by movy** [lesgrieve/paste-indent#2]
First of all, thanks for the extension, it's a real life-saver with python!  I just came from auto-formatted and perfectly-linted JS world and pythonic manual white-space formatting is such a mess tbh, cannot imagine doing it w/o paste-indent.

I use remote workspaces with VSCode, and paste-indent causes a little lag when pasting, which is okay, but the real problem happens when there's even a slightest connectivity issue: pasting sequences are completely 'swallowed', but once connectivity is restored, some of the 'swallowed' paste-indent commands are executed and I get several copies of my clipboard pasted. 
Normal VSCode's paste command, typing etc. works fine during such periods, as VSCode somehow buffers key inputs locally and all connection problems go unnoticed, but not with paste-indent unfortunatelly.  This is easy to simulate by setting up remote workspace, open a document and going offline, you'll have 15-30 seconds to experiment till VSCore realizes connection is lost and attempts to re-connect.

This extension significantly improves quality of life with Python, and I'd be grateful if you could look into it and see if it's fixable. 

Thanks heaps!

**Comment by mikelei8291** [comment]
That's because the extension doesn't specify its preferred extension host location in its `package.json` file, so VS Code would default to run it at the remote extension host, causing the lag. It must specify the preferred extension host location with `"extensionKind": ["ui", "workspace"]` to make it run in the local extension host, and when that's not available, on the remote extension host (e.g. VS Code for the Web). Otherwise, it has to also be a [web extension](https://code.visualstudio.com/api/extension-guides/web-extensions) to be working in the web browser.

Documentation: https://code.visualstudio.com/api/advanced-topics/extension-host#preferred-extension-location

However, since this repo hasn't been updated for a while, you can implement this workaround in your VS Code: https://code.visualstudio.com/api/advanced-topics/remote-extensions#incorrect-execution-location

Add the following setting in the `settings.json` file of VS Code:
```json
"remote.extensionKind": {
    "lesgrieve.paste-indent": ["ui"]
}
```

**Comment by movy** [comment]
trying this now, thanks so much for clarification!

**Comment by lesgrieve** [comment]
This issue has been addressed in [43222ed](https://github.com/lesgrieve/paste-indent/commit/43222edef53dd4f8da4539a3fd06982c19731567) and published to https://marketplace.visualstudio.com/items?itemName=LesGrieve.paste-indent as version 0.0.2. Please confirm this issue is resolved.

**Comment by lesgrieve** [comment]
> That's because the extension doesn't specify its preferred extension host location in its `package.json` file, so VS Code would default to run it at the remote extension host, causing the lag. It must specify the preferred extension host location with `"extensionKind": ["ui", "workspace"]` to make it run in the local extension host, and when that's not available, on the remote extension host (e.g. VS Code for the Web). Otherwise, it has to also be a [web extension](https://code.visualstudio.com/api/extension-guides/web-extensions) to be working in the web browser.
> 
> Documentation: https://code.visualstudio.com/api/advanced-topics/extension-host#preferred-extension-location
> 
> However, since this repo hasn't been updated for a while, you can implement this workaround in your VS Code: https://code.visualstudio.com/api/advanced-topics/remote-extensions#incorrect-execution-location
> 
> Add the following setting in the `settings.json` file of VS Code:
> 
> ```json
> "remote.extensionKind": {
>     "lesgrieve.paste-indent": ["ui"]
> }
> ```

Thank you @mikelei8291 for doing the work for me ;)
I stopped getting notifications from GitHub and was not aware of this issue.
